tools/toolchain/clang_lto_pgo.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash -uex
+
+if [ -z "${CLANG_BUILD}" ]; then
+    echo "Skip building optimized clang"
+    exit 0
+elif [ "${CLANG_BUILD}" = "SKIP" ]; then
+    echo "Skip building optimized clang"
+    exit 0
+elif [ "${CLANG_BUILD}" = "BUILD" ]; then
+    echo "build optimized clang"
+elif [ "${CLANG_BUILD}" = "INSTALL" ]; then
+    echo "build and install optimized clang"
+elif [ "${CLANG_BUILD}" = "INSTALL_FROM" ]; then
+    echo "install prebuilt optimized clang"
+else
+    echo "Not sure what to do with ${CLANG_BUILD}"
+    exit 1
+fi
+
+if [ -z "${CLANG_ARCH}" ]; then
+    echo "CLANG_ARCH is not defined"
+elif [ "${CLANG_ARCH}" != "amd64" ]; then
+    LLVM_TARGET_ARCH=AArch64
+    LLVM_TARGET="-DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=AArch64;WebAssembly"
+else
+    LLVM_TARGET_ARCH=X86
+    LLVM_TARGET="-DLLVM_EXPERIMENTAL_TARGETS_TO_BUILD=WebAssembly"
+fi
+
+mkdir -p /scylladb/clang_build
+DIR="/scylladb/clang_build"
+
+CLANG_BIN=$DIR/clang/build/bin
+
+# Which LLVM release to build in order to compile Scylla
+LLVM_CLANG_TAG=17.0.6
```

Comment:
We're diverging from Fedora here. It means building might only work from the toolchain and not from plain Fedora (of course one can build clang, but it's nice if plain Fedora can be used too).

Reply:
Do you mean we should use 16.0.6 (same version as our frozen toolchain currently provides)?

Reply:
if we adhere to this guideline -- to avoid diverging from f38's clang/llvm packaging. (BTW, f38 uses different src packages for llvm and clang packages), we would need to apply the patches here: https://src.fedoraproject.org/rpms/clang/commits/f38 . there are no strong evidences that we have existing FTBFS caused by the absence of either of these two patches. the first one was intended to address a https://github.com/scylladb/seastar/issues/1759, but it turned out it was a bug in seastar, and was fixed by https://github.com/scylladb/seastar/commit/7fe5bdd6d85b6f3a57491882a4fbd809f54956f0. the other was created to address https://github.com/scylladb/seastar/issues/1772 . this is a genuine bug of Clang. it only surfaces when compiling with the combination of Clang, C++23 and GCC-13. so we might not need it, as we are using C++20 at this moment.  when we switch to C++23, we would have the luxury to bump up the dbuild base image to f39 or up, which has Clang-17 or up. by then, we can apply this patch. yes, i also applied this patch in f39.

Reply:
but my 2 cent is, we might need to use clang 18 here instead of clang 16 or clang 17. as the latter two are not maintained by llvm upstream anymore. if one fail to build scylla with clang 16 from f38 or clang 17 from f39, it's likely a bug in older clang not one from scylla.